signac/contrib/project.py

```diff
@@ -87,6 +89,16 @@ def _resolve_statepoint_filter(self, q):
             else:
                 yield 'statepoint.{}'.format(k), v
 
+    def _find_job_ids(self, filter=None, doc_filter=None):
+        if filter:
+            filter = dict(self._resolve_statepoint_filter(filter))
+            if doc_filter:
+                filter.update(doc_filter)
+        elif doc_filter:
+            filter = doc_filter
+        return self._collection._find(filter)
+
+    @deprecated(deprecated_in="1.3", removed_in="2.0", current_version=__version__)
```

Comment:
The `JobSearchIndex` class is not explicitly exposed as part of the public API, but only implicitly as a return value of `Project.build_job_search_index()`. Since that function is now deprecated, I don't think we need to deprecate the function as part of this class.